src/modules/twap/services/createTwapOrderTxs.ts

```diff
@@ -0,0 +1,60 @@
+import { IConditionalOrder } from 'abis/types/ComposableCoW'
+import { defaultAbiCoder } from '@ethersproject/abi'
+import { MetaTransactionData } from '@safe-global/safe-core-sdk-types'
+import { TWAPOrder, TWAPOrderStruct } from '../types'
+import { TWAP_HANDLER_ADDRESS, TWAP_ORDER_STRUCT } from '../const'
+import { COMPOSABLE_COW_ADDRESS } from '../../advancedOrders/const'
+import { hexZeroPad } from '@ethersproject/bytes'
+import { TwapOrderCreationContext } from '../hooks/useTwapOrderCreationContext'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+function getTwapOrderParamsStruct(
+  chainId: SupportedChainId,
+  order: TWAPOrder
+): IConditionalOrder.ConditionalOrderParamsStruct {
+  const twapOrderData: TWAPOrderStruct = {
+    sellToken: order.sellAmount.currency.address,
+    buyToken: order.buyAmount.currency.address,
+    receiver: order.receiver,
+    partSellAmount: order.sellAmount.divide(order.numOfParts).quotient.toString(),
+    minPartLimit: order.buyAmount.divide(order.numOfParts).quotient.toString(),
+    t0: order.startTime,
+    n: order.numOfParts,
+    t: order.timeInterval,
+    span: order.span,
+  }
+
+  return {
+    handler: TWAP_HANDLER_ADDRESS[chainId],
+    salt: hexZeroPad(Buffer.from(Date.now().toString(16), 'hex'), 32),
+    staticInput: defaultAbiCoder.encode([TWAP_ORDER_STRUCT], [twapOrderData]),
+  }
+}
+
+export function createTwapOrderTxs(order: TWAPOrder, context: TwapOrderCreationContext): MetaTransactionData[] {
+  const { chainId, composableCowContract, needsApproval, erc20Contract, spender } = context
+
+  const sellTokenAddress = order.sellAmount.currency.address
+  const sellAmountAtoms = order.sellAmount.quotient.toString()
+
+  // TODO: support other conditional orders (stop loss, GAT, etc.)
+  const creationParams = getTwapOrderParamsStruct(chainId, order)
+
+  const createOrderTx = {
+    to: COMPOSABLE_COW_ADDRESS[chainId],
+    data: composableCowContract.interface.encodeFunctionData('create', [creationParams, true]),
```

Comment:
Suggestion for a different way of creating the tx data, which uses the types and parameters more in a TS fashion https://github.com/cowprotocol/cowswap/blob/ed50eb839629d7df29a404a5d4663b56330d1900/src/modules/operations/bundle/buildPresignTx.ts#L12-L11

Reply:
Yes, I'm aware of it. I don't have a strong opinion on it,but there are some differences:
 - `encodeFunctionData` returns `string`
 - `populateTransaction` returns `Promise<string>`
 - `encodeFunctionData` has type overloads, so, the interface consistency is safe
 
<img width="513" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/c8df329c-d8de-42de-bf4e-22d8e8f86372">

Reply:
Cool, didn't know
I don't mind also to use this version.

Then, do you think we should update the method I linked?

Reply:
Just to make it synchronous - yes, it make sense